EIPS/eip-5027.md

```diff
@@ -0,0 +1,84 @@
+---
+eip: 5027
+title: Unlimit contract code size
+author: Qi Zhou @qizhou
+discussions-to: https://ethereum-magicians.org/t/eip-5027-unlimit-contract-code-size/9010
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-04-21
+requires: EIP-170
+---
+
+
+## Abstract
+
+Unlimit the contract code size so that users can deploy a large-code contract without worrying about splitting the contract into several sub-contracts.
+
+
+## Motivation
+
+The motivation is to
+- unlimit the code size, i.e., only limited by block gas limit;
+- minimize the changes to existing code; and
+- address the possible attack vectors in multiple contract-code-related opcodes including
+
+`CODESIZE/CODECOPY/EXTCODESIZE/EXTCODECOPY/EXTCODEHASH/DELEGATECALL/CALL/CALLCODE/STATICCALL/CREATE/CREATE2`.
```

Comment:
The abstract is to give a summary of the EIP, which you have currently put in the motivation. The motivation is to rationalize the EIP - in this case, the reason ("so that users can deploy a large-code contract without worrying about splitting the contract into several sub-contracts") is put in the abstract and rationale instead.

Reply:
Thanks for the comment.  I have moved this part to Abstract and moved the reason  "so that users can deploy a large-code contract without worrying about splitting the contract into several sub-contracts" here.  In addition, I have moved the first two paragraphs in Rationale to here.  Please take a look.